mesonbuild/build.py

```diff
@@ -1155,12 +1155,12 @@ def process_kwargs(self, kwargs):
             # was not specified and win_subsystem should be used instead.
             self.gui_app = None
             if kwargs.get('gui_app') is not None:
-                if 'win_subsystem' in kwargs:
+                if kwargs.get('win_subsystem') is not None:
                     raise InvalidArguments('Can specify only gui_app or win_subsystem for a target, not both.')
                 self.gui_app = kwargs['gui_app']
                 if not isinstance(self.gui_app, bool):
                     raise InvalidArguments('Argument gui_app must be boolean.')
-            self.win_subsystem = self.validate_win_subsystem(kwargs.get('win_subsystem', 'console'))
+            self.win_subsystem = kwargs.get('win_subsystem', 'console')
```

Comment:
If we can guarantee that win_subsystem is valid, then that means it always passed through typed_kwargs and we also don't need to use .get().

But I don't think we can make that guarantee because of e.g. modules.

Reply:
Yeah, exactly. I'd like to eventually get to the point where the initializers for the various Target classes are all fully typed, then we would be able to guarantee that they're valid (or at least validly typed) and could do away with this. I'm also trying to land these changes at all, so starting small

Reply:
What I mean is that for now it seems like we should probably just bite the bullet and check it twice.

Reply:
There's only one call to create an Executable that I can find in any modules, which is in the rust module. How would you feel about adding the keyword there and `assert 'win_subsystem' in kwargs` until we move to fully typed args?

Reply:
Hmm, actually on second thought the rust usage of it doesn't permit that kwarg anyway, so we shouldn't need to check it. We can either specially initialize it as None or just use .get here, but either way it looks like in the end we actually do not have to validate its contents.

So I guess we do not have to do anything here. Sorry for the confusion.

Reply:
Does that mean that you don't have any more objections? I did fix the `plumb` typo already